scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/FilterEvaluator.java

```diff
@@ -444,22 +443,20 @@ private Iterable<JsonNode> getCandidateNodes(@NotNull final Path path,
    */
   private boolean isEmpty(@NotNull final JsonNode node)
   {
-    if (node.isArray())
+    if (!node.isArray())
     {
-      Iterator<JsonNode> iterator = node.elements();
-      while (iterator.hasNext())
-      {
-        if (!isEmpty(iterator.next()))
-        {
-          return false;
-        }
-      }
-      return true;
+      return node.isNull();
     }
-    else
+
+    for (JsonNode jsonNode : node)
     {
-      return node.isNull();
+      if (!isEmpty(jsonNode))
```

Comment:
Honestly, the original version of this logic is a little baffling to parse, but as far as I can tell this is logically identical so it's fine.